cEP-0019.md

```diff
@@ -0,0 +1,467 @@
+
+GitLab is beyond the scope of this project, as GitHub and GitLab have quite
+different APIs and most core projects of coala are based on GitHub. This
+proposal will only discuss reactions on GitHub, but not award emoji on GitLab.
+
+There are 6 types of reactions on GitHub: CONFUSED 😕, HEART ❤️, HOORAY 🎉,
+LAUGH 😄, THUMBS_DOWN 👎, THUMBS_UP 👍.
+
+There are three types of reviews on a pull request:
+
+1. Issue Comment
+   [see sample](https://github.com/coala/gh-board/pull/18#issuecomment-385878735)
+
+2. Review Summary
+   [see sample](https://github.com/coala/gh-board/pull/18#pullrequestreview-116783753)
+
+3. Review Discussion
+   [see sample](https://github.com/coala/gh-board/pull/18#discussion_r186249423)
+
+However, those reactions spread over all PRs and are not collected and
+analyzed. Moreover, GitHub doesn't send notifications to reviewers who receive
+reactions. By tracking reactions, a nice feedback loop would be created
+for both senior and junior reviewers.
+
+## Introduction
+
+The meta-review project is about making use of reactions to build a
+meta-review system, similar to a meta-moderation system. These responses
+are to be collected, processed and displayed on
+[gh-board](https://github.com/coala/gh-board), which is a nice kanban
+board. Compared to [Huboard](https://huboard.com/) which requires server-side
+support, gh-board is serverless.
+
+The scope of this project would exclude non-public meta-reviews. That is,
+there would not be any anonymous meta-reviews as that introduces too many
+additional complexities. The public only version would be used for
+a long time before adding support for non-public meta-reviews.
+
+Note there is a technical limitation in that reactions can not be placed
+onto the "review summary", which means general review comments should not
+be put into the "review summary" text box, as it will be not able to be
+meta-reviewed.
+
+## Objectives
+
+One goal of this project is to encourage people to do meta-reviews,
+systematically. Meta-reviews in need will be displayed on gh-board. The
+implementation details are discussed in the next section.
+
+Also, to encourage people to do more meta-reviews, statistics of meta-reviews
+are to be collected and analysed. People who do meta-reviews will get scores
+according to some metrics. A ranking list is to be displayed on the community
+site. The newcomer guide will also be updated to require newcomers to finish
+at least one meta-review and receive at least one positive meta-review.
+
+To conclude, this project builds a meta-review system on the community site
+and gh-board to display score ranking statistics and meta-reviews in need,
+respectively, and encourage people to do meta-review.
+
+## Meanings of reactions
+
+As mentioned in the background section, GitHub provides 6 types of reactions.
+They have different meanings in the context of meta-review system.
+
+There are 6 types of reactions on GitHub: CONFUSED 😕, HEART ❤️, HOORAY 🎉,
```

Comment:
dumping random thoughts here:


Currently we've only given meaning to THUMBS_DOWN & THUMBS_UP.

I guess LAUGH is not able to be easily used for meta-review.

CONFUSED on review comments has a fairly obvious meaning, but how to we integrate it into this meta-review system...?  Is it a neutral score, suggesting clarifications are needed, or does it have a negative score , and is it more negative than THUMBS_DOWN or less negative?  i.e. the review comment wasnt very useful...?  Or we might decide it means that the CONFUSED is UNHELPFUL/WTF while THUMBS_UP/THUMBS_DOWN is more like a vote and thus THUMBS_DOWN is merely NOT_PREFERRED (i.e. it was a useful review comment, but there was a more useful review comment / more appropriate solution found)

HOORAY on review comments possibly has a few usages, and might need more community input on what meanings it should have, and how it should be part of meta-review.

HEART on review comments possibly has a few usages, but it feels like it is best used for showing a greater amount of appreciation for a review comment than THUMBS_UP.